File: 최승훈/2주차/10845.py
from sys import stdin

N = int(stdin.readline())
queue = []
for i in range(N):
    A = stdin.readline().split() 
    # sys.stdin.readline().split()
    
    if A[0] == 'push': 
        queue.append(A[1])

    elif A[0] == 'pop': 
        if queue:
            print(queue.pop(0))
        else: 
            print(-1)

    elif A[0] == 'size': 
        print(len(queue))

    elif A[0] == 'empty':
        if len(queue) == 0: 
            print(1)
        else: 
            print(0)
            
    elif A[0] == 'front':
        if len(queue) == 0: 
            print(-1)
        else: 
            print(queue[0])
    
    elif A[0] == 'back':
        if len(queue) == 0: 
            print(-1)
        else: 
            print(queue[-1])

File: 권현욱/2주차/10845.cpp
#include <iostream>
#include <queue>
#include <string>

using namespace std;

int main()
{
    queue<int> q;
    string str;

    int N;
    cin >> N;

    for (int i = 0; i < N; i++)
    {
        cin >> str;

        if (str == "push")
        {
            int X;
            cin >> X;
            q.push(X);
        }
        else if (str == "pop")
        {
            if (q.empty())
                cout << "-1" << "\n";
            else
            {
                cout << q.front() << "\n";
                q.pop();
            }
        }
        else if (str == "size")
        {
            cout << q.size() << "\n";
        }
        else if (str == "empty")
        {
            if (q.empty())
                cout << "1" << "\n";
            else
                cout << "0" << "\n";
        }
        else if (str == "front")
        {
            if (q.empty())
                cout << "-1" << "\n";
            else    
                cout << q.front() << "\n";
        }
        else if (str == "back")
        {
            if (q.empty())
                cout << "-1" << "\n";
            else
                cout << q.back() << "\n";
        }
    }

    return 0;
}
